Unable to install due to errors in compilation

**Describe the bug**
Upon running `npm install zeromq@6.0.0-beta.11` I end up getting a whole bunch of errors.


**Tested on**
 - OS: Mac M1
 - ZeroMQ.js version: 6.0.0-beta.11

I'm assuming this is something to do with a setup at my end, if that's the case please feel free to close this issue.

[2022-11-25T10_44_12_769Z-debug-0.log](https://github.com/zeromq/zeromq.js/files/10092357/2022-11-25T10_44_12_769Z-debug-0.log)

<details>
  <summary>Here are the logs</summary>
  
```
205 error          ^
205 error /Users/donjayamanne/temp/sampleJS/node_modules/zeromq/build/libzmq-staging/zeromq-4.3.4/src/object.hpp:149:18: note: overridden virtual function is here
205 error     virtual void process_conn_failed ();
205 error                  ^
205 error In file included from /Users/donjayamanne/temp/sampleJS/node_modules/zeromq/build/libzmq-staging/zeromq-4.3.4/src/stream_listener_base.cpp:34:
205 error In file included from /Users/donjayamanne/temp/sampleJS/node_modules/zeromq/build/libzmq-staging/zeromq-4.3.4/src/zmtp_engine.hpp:44:
205 error /Users/donjayamanne/temp/sampleJS/node_modules/zeromq/build/libzmq-staging/zeromq-4.3.4/src/stream_engine_base.hpp:76:10: warning: 'out_event' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
205 error     void out_event ();
205 error          ^
205 error /Users/donjayamanne/temp/sampleJS/node_modules/zeromq/build/libzmq-staging/zeromq-4.3.4/src/io_object.hpp:73:10: note: overridden virtual function is here
205 error     void out_event () ZMQ_OVERRIDE;
205 error          ^
205 error 2 warnings generated.
205 error ld: library not found for -lsodium
205 error clang: error: linker command failed with exit code 1 (use -v to see invocation)
205 error make: *** [Release/zeromq.node] Error 1
205 error gyp ERR! build error 
205 error gyp ERR! stack Error: `make` failed with exit code: 2
205 error gyp ERR! stack     at ChildProcess.onExit (/Users/donjayamanne/.nvm/versions/node/v16.14.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
205 error gyp ERR! stack     at ChildProcess.emit (node:events:526:28)
205 error gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
205 error gyp ERR! System Darwin 22.1.0
205 error gyp ERR! command "/Users/donjayamanne/.nvm/versions/node/v16.14.2/bin/node" "/Users/donjayamanne/.nvm/versions/node/v16.14.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
205 error gyp ERR! cwd /Users/donjayamanne/temp/sampleJS/node_modules/zeromq
205 error gyp ERR! node -v v16.14.2
205 error gyp ERR! node-gyp -v v9.0.0
205 error gyp ERR! not ok
206 verbose exit 1
207 timing npm Completed in 67590ms
208 verbose unfinished npm timer reify 1669373052861
209 verbose unfinished npm timer reify:build 1669373053358
210 verbose unfinished npm timer build 1669373053358
211 verbose unfinished npm timer build:deps 1669373053358
212 verbose unfinished npm timer build:run:install 1669373053362
213 verbose unfinished npm timer build:run:install:node_modules/zeromq 1669373053362
214 verbose code 1
215 error A complete log of this run can be found in:
215 error     /Users/donjayamanne/.npm/_logs/2022-11-25T10_44_12_769Z-debug-0.log
```

</details>

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -44,7 +44,6 @@ jobs:
           #   ARCH: arm64
           #   cpp_arch: amd64_arm64
           #   zmq_draft: false
-          #   zmq_version: "v4.3.4"
 
           - os: macos-11
             node_version: 18

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -25,8 +25,6 @@ jobs:
           - x64
         zmq_draft:
           - false
-        zmq_version:
-          - 4.3.4
         docker:
           - ""
         docker_cmd:
@@ -39,7 +37,6 @@ jobs:
             ARCH: x86
             cpp_arch: amd64_x86
             zmq_draft: false
-            zmq_version: 4.3.4
 
           # - os: windows-2022
           #   node_version: 18
@@ -47,7 +44,7 @@ jobs:
           #   ARCH: arm64
           #   cpp_arch: amd64_arm64
           #   zmq_draft: false
-          #   zmq_version: 4.3.4
+          #   zmq_version: "v4.3.4"
 
           - os: macos-11
             node_version: 18
@@ -55,7 +52,6 @@ jobs:
             ARCH: x86_64
             cpp_arch: x64
             zmq_draft: false
-            zmq_version: 4.3.4
 
           - os: macos-11
             node_version: 18
@@ -63,7 +59,6 @@ jobs:
             ARCH: arm64
             cpp_arch: amd64_arm64
             zmq_draft: false
-            zmq_version: 4.3.4
 
           - os: ubuntu-22.04
             docker: node:18-alpine
@@ -73,10 +68,8 @@ jobs:
             ARCH: x64
             cpp_arch: x64
             zmq_draft: false
-            zmq_version: 4.3.4
 
     env:
-      ZMQ_VERSION: ${{ matrix.zmq_version }}
       ZMQ_DRAFT: ${{ matrix.zmq_draft }}
       ZMQ_SHARED: false
       ARCH: ${{ matrix.ARCH }}

--- a/script/build.ts
+++ b/script/build.ts
@@ -8,7 +8,7 @@ function main() {
   const zmq_rev =
     // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions
     process.env.ZMQ_VERSION || "4e193f36fc7d0f729a7c87d55fff18d8c0db5ebf"
-  const src_url = `"https://github.com/zeromq/libzmq/archive/${zmq_rev}.tar.gz`
+  const src_url = `https://github.com/zeromq/libzmq/archive/${zmq_rev}.tar.gz`
 
   const libzmq_build_prefix = `${root}/build/libzmq-staging`
   const libzmq_install_prefix = `${root}/build/libzmq`

--- a/script/build.ts
+++ b/script/build.ts
@@ -7,7 +7,7 @@ const root = dirname(__dirname)
 function main() {
   const zmq_rev =
     // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions
-    process.env.ZMQ_VERSION || "4e193f36fc7d0f729a7c87d55fff18d8c0db5ebf"
+    process.env.ZMQ_VERSION || "20de92ac0a2b2b9a1869782a429df68f93c3625e"
   const src_url = `https://github.com/zeromq/libzmq/archive/${zmq_rev}.tar.gz`
 
   const libzmq_build_prefix = `${root}/build/libzmq-staging`

--- a/script/build.ts
+++ b/script/build.ts
@@ -5,9 +5,10 @@ import {mkdir, cd, exec, find, mv} from "shelljs"
 const root = dirname(__dirname)
 
 function main() {
-  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions
-  const zmq_version = process.env.ZMQ_VERSION || "4.3.4"
-  const src_url = `https://github.com/zeromq/libzmq/releases/download/v${zmq_version}/zeromq-${zmq_version}.tar.gz`
+  const zmq_rev =
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions
+    process.env.ZMQ_VERSION || "4e193f36fc7d0f729a7c87d55fff18d8c0db5ebf"
+  const src_url = `"https://github.com/zeromq/libzmq/archive/${zmq_rev}.tar.gz`
 
   const libzmq_build_prefix = `${root}/build/libzmq-staging`
   const libzmq_install_prefix = `${root}/build/libzmq`
@@ -16,8 +17,8 @@ function main() {
     process.platform === "win32" ? ".lib" : ".a"
   }`
 
-  const src_dir = `zeromq-${zmq_version}`
-  const tarball = `zeromq-${zmq_version}.tar.gz`
+  const src_dir = `libzmq-${zmq_rev}`
+  const tarball = `libzmq-${zmq_rev}.tar.gz`
 
   // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions
   const CMAKE_BUILD_TYPE = process.env.CMAKE_BUILD_TYPE || "Release"

